hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BPServiceActor.java

```diff
@@ -258,7 +258,7 @@ NamespaceInfo retrieveNamespaceInfo() throws IOException {
     while (shouldRun()) {
       try {
         nsInfo = bpNamenode.versionRequest();
-        LOG.debug(this + " received versionRequest response: " + nsInfo);
+        LOG.debug("{} received versionRequest response: {}", this, nsInfo);
```

Comment:
HI, IMO,  "if (LOG.isDebugEnabled()) {...}" is better.

Reply:
```java
if (LOG.isDebugEnabled()) {
    LOG.debug("{} received versionRequest response: {}", this, nsInfo);
}

```

Reply:
LOG.debug already does isDebugEnabled() internally, the original issue is String concatenation with heavy object.

This change is sufficient IMO.

Reply:
> HI, IMO, "if (LOG.isDebugEnabled()) {...}" is better.

Thanks for your comment. I agree with @ayushtkn and @virajjasani here. LOG.debug already does isDebugEnabled() internally.

Reply:
> > HI, IMO, "if (LOG.isDebugEnabled()) {...}" is better.
> 
> Thanks for your comment. I agree with @ayushtkn and @virajjasani here. LOG.debug already does isDebugEnabled() internally. <img alt="image" width="872" src="https://private-user-images.githubusercontent.com/55134131/369738260-ed4f9b82-8dbd-4a89-80b5-1263f982c782.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc0MjQ4NzYsIm5iZiI6MTcyNzQyNDU3NiwicGF0aCI6Ii81NTEzNDEzMS8zNjk3MzgyNjAtZWQ0ZjliODItOGRiZC00YTg5LTgwYjUtMTI2M2Y5ODJjNzgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI3VDA4MDkzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViMTllNTJmYWZiN2I5NjMxZDUyMDNlNmZiZGUyMGZlNjRiMjRmNDJjNGVmNTIwYjFmNmI3ZTI5ODA4OTM2MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-iBwFzkAvy6JJOqsn_2Mpwe9MCFUgmJCNj6wSx7p1tk">
> 

@tomscut Got it, thanks, LGTM.